src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -26,9 +26,34 @@
 
 import org.spongepowered.api.CatalogType;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the ascending direction, if this rail direction is ascending.
+     *
+     * @return The ascending direction, if possible
+     */
+    Optional<Direction> getAscendingDirection();
+
+    /**
+     * Returns the first direction of this rail direction.
+     *
+     * @return The first direction
+     * @see RailDirection#getSecondDirection()
+     */
+    Direction getFirstDirection();
+
+    /**
+     * Returns the second direction of this rail direction.
+     *
+     * @return The second direction
+     * @see RailDirection#getFirstDirection()
+     */
+    Direction getSecondDirection();
```

Comment:
Maybe add a method like `isFacing(Direction) : boolean` to avoid the trouble to check both direction components in the own code.

Reply:
Good idea! I also thought of adding something like this here, this will really reduce a condition code in a plugin.